internal/trie/node/decode.go

```diff
@@ -78,18 +83,14 @@ func decodeBranch(reader io.Reader, header byte) (node *Node, err error) {
 
 	sd := scale.NewDecoder(reader)
 
-	nodeType := header >> 6
-	if nodeType == branchWithValueHeader {
-		var value []byte
-		// branch w/ value
-		err := sd.Decode(&value)
+	if variant == branchWithValueVariant.bits {
+		err := sd.Decode(&node.Value)
 		if err != nil {
 			return nil, fmt.Errorf("%w: %s", ErrDecodeValue, err)
 		}
-		node.Value = value
 	}
 
-	for i := 0; i < 16; i++ {
+	for i := 0; i < ChildrenCapacity; i++ {
```

Comment:
OOS: seems odd that `ChildrenCapacity` is exported

Reply:
It has been exported previously (from a suggestion of Eclesio as I recall);
It's because it's used by the `lib/trie` package to iterate over children.
We could define as a local constant in both packages, but yeah not a big deal imo.